examples/ethereum/ethereum.go

```diff
@@ -42,31 +81,26 @@ func (c *EthereumConnector) Start() {
 				log.Fatal().Err(err)
 			case header := <-headers:
 				// Header doesn't contain full block information, so get block
-				block, err := client.BlockByHash(context.Background(), header.Hash())
+				block, err := c.client.BlockByHash(context.Background(), header.Hash())
 				if err != nil {
 					log.Fatal().Err(err).Msg("BlockByHash error")
 				}
 
-				PrintBlock(block)
+				LogBlock(block)
 
 				// EthBlock -> Block -> Protobuf -> kafka
 				var blockData Block
 				blockData.UnmarshalEthBlock(block)
-				err = c.ProduceAndCommitMessage("ethereum", "ethereum", &blockData)
-				if err != nil {
-					log.Error().Err(err).Msg("Kafka write proto")
-				}
+
+				c.EventSink <- buildKafkaMsg(kafkautils.MsgTypeFct, &blockData)
```

Comment:
Let's use transactions and function calls to send kafka messages instead of chan

Reply:
I think it'd be better to let `connector` lib handle the kafka interaction, so they wouldn't use it in the wrong way.